packages/phasor/src/elements/surface-element.ts

```diff
@@ -13,6 +13,11 @@ export interface ISurfaceElement {
   index: string;
   seed: number;
 }
+export type SurfaceElementKey = keyof ISurfaceElement;
+export interface ISurfaceElementView extends ISurfaceElement {
+  display: boolean;
```

Comment:
I don't feel right about making these two properties local-only. What if they need to be collaboratively shared?

So I suggest not to introduce this _view_ concept. Instead, what I think about is introducing concepts like _LocalElementModifier_s that can "shadow" CRDT state field by local ones. Each modifier is identified by an element id, and its fields have higher priority than CRDT element props when an element is being rendered.

/cc @Flrande

Reply:
now making these two properties local-only is because we haven't use these two properties in today's functionality.(
for exmaple, we havn't support that element can change its opacity, but eventually it will I believe.)
So Ii it needs to be collaborative, then we just add them to crdt model. And the view interface inherits the model interface.

Reply:
Yes, what we need is to "shadow" CRDT field.

Reply:
One simple approach I can think of is to maximize the shareability of states, so that transparency, which does not affect interactions and only affects visual effects, can be directly incorporated into CRDT?

Reply:
It seems that it would be more reasonable to place the display field I previously added only within the TextElement. Instead of using a public field for the surface to control the rendering of this element, I should directly control the text rendering through the render function in TextElement.